include/crpropa/massdistribution/Density.h

```diff
@@ -0,0 +1,47 @@
+#ifndef CRPROPA_DENSITY_H
+#define CRPROPA_DENSITY_H
+
+#include "crpropa/Units.h"
+#include "crpropa/Vector3.h"
+#include "crpropa/Referenced.h"
+
+namespace crpropa {
+/**
+ @class Density
+ @brief Abstract base class for targetdensity
+ */
+class Density: public Referenced {
+ public:
+  virtual ~Density() {
+  }
+  virtual double getDensity(const Vector3d &position) const { 	// sum of all densities
```

Comment:
This could cause an error, because not all models have each hydrogyn component (e.g. Cordes). 
If we define the pure virtual functions as "=0", it is necessary to define in each model a "return 0" for not used components. I think it is easier to define the standard virtual funktion with "return 0" in the base class and only overwrite the functions, which are implemented in the models.

Reply:
OK. That is a valid reason. However, it still looks a bit complicated solution, but since I don't have any better suggestion for now, this can go. When we will have the wider picture and more practical experience we can re-implement it.